app/javascript/layouts/app_router/components/AppRouter__Header.res

```diff
@@ -0,0 +1,200 @@
+          className="inline-block object-contain rounded-full text-tiny" src alt={User.name(user)}
+        />,
+    )}
+  </button>
+}
+
+let renderLinks = user => {
+  <Dropdown selected={selected(user)} contents={links()} right=true key="links-dropdown" />
+}
+
+let headerLink = (key, link) =>
+  <div
+    key
+    className="md:ml-2 text-sm font-semibold text-center cursor-default flex w-1/2 sm:w-1/3 md:w-auto justify-center">
+    <a
+      className="no-underline bg-gray-100 md:bg-white hover:bg-gray-200 text-gray-900 rounded-lg hover:text-primary-500 w-full p-4 md:px-3 md:py-2"
+      href={link->School.linkUrl}
+      target=?{School.localLinks(link) ? None : Some("_blank")}
+      rel=?{School.localLinks(link) ? None : Some("noopener")}>
+      {School.linkTitle(link)->str}
+    </a>
+  </div>
+
+let signOutLink = () =>
+  <div
+    key="Logout-button"
+    className="md:ml-2 text-sm font-semibold cursor-default flex w-1/2 sm:w-1/3 md:w-auto justify-center">
+    <div className="flex items-center justify-center">
+      <a
+        href="/users/sign_out"
+        rel="nofollow"
+        className="border border-primary-500 rounded px-2 py-1 text-primary-500 text-xs md:text-sm md:leading-normal m-4 md:m-0 no-underline font-semibold">
+        <FaIcon classes="fas fa-power-off" /> <span className="ml-2"> {t("sign_out")->str} </span>
+      </a>
+    </div>
+  </div>
+
+let signInLink = () =>
+  <div
+    key="SignIn-button"
+    className="md:ml-2 text-sm font-semibold cursor-default flex w-1/2 sm:w-1/3 md:w-auto justify-center">
+    <div className="flex items-center justify-center">
+      <a
+        className="border border-primary-500 rounded px-2 py-1 text-primary-500 text-xs md:text-sm md:leading-normal m-4 md:m-0 no-underline font-semibold"
+        href="/users/sign_in">
+        <FaIcon classes="fas fa-power-off" /> <span className="ml-2"> {t("sign_in")->str} </span>
+      </a>
+    </div>
+  </div>
+
+let notificationButton = hasNotifications =>
+  <Notifications__Root
+    key="notifications-button"
+    wrapperClasses="relative md:ml-1 pt-1 md:pt-0 text-sm font-semibold cursor-default flex w-8 h-8 md:w-9 md:h-9 justify-center items-center rounded-lg hover:bg-gray-200"
+    iconClasses="app-router-header__notifications-unread-bullet"
+    buttonClasses="font-semibold text-gray-900 hover:text-primary-500 w-full flex items-center justify-center "
+    icon="if i-bell-regular text-xl"
+    hasNotifications
+  />
+
+let isMobile = () => Webapi.Dom.Window.innerWidth(Webapi.Dom.window) < 768
+
+let headerLinks = (links, isLoggedIn, user, hasNotifications) => {
+  let (visibleLinks, dropdownLinks) = switch (Array.to_list(links), isMobile()) {
+  | (links, true) => (Array.of_list(links), [])
+  | (list{l1, l2, l3, l4, l5, ...rest}, false) => (
+      [l1, l2, l3],
+      Js.Array.concat(Array.of_list(rest), [l4, l5]),
+    )
+  | (fourOrLessLinks, false) => (Array.of_list(fourOrLessLinks), [])
+  }
+
+  switch visibleLinks {
```

Comment:
Not seen this pattern before. Just curious why do we need a switch here on an array here?